lib/OrdinaryDiffEqNewmark/src/newmark_perform_step.jl

```diff
@@ -0,0 +1,118 @@
+function initialize!(integrator, cache::NewmarkBetaCache)
+    duprev, uprev = integrator.uprev.x
+    integrator.f(cache.fsalfirst, integrator.uprev, integrator.p, integrator.t)
+    integrator.stats.nf += 1
+    integrator.fsalfirst = cache.fsalfirst
+    integrator.fsallast = cache.fsalfirst
+    # integrator.fsallast = du_alias_or_new(cache.nlsolver, integrator.fsalfirst)
+    return
+end
+
+@muladd function perform_step!(integrator, cache::NewmarkBetaCache, repeat_step = false)
+    @unpack t, dt, f, p = integrator
+    @unpack β, γ, nlcache = cache
+
+    M = f.mass_matrix
+
+    # Evaluate predictor
+    aₙ = integrator.fsalfirst.x[1]
+    vₙ, uₙ = integrator.uprev.x
+
+    evalcache = NewmarkDiscretizationCache(
+        f, t, p,
+        dt, β, γ,
+        aₙ, vₙ, uₙ
+    )
+    SciMLBase.reinit!(nlcache, aₙ, p = evalcache)
+    solve!(nlcache)
+    if nlcache.retcode != ReturnCode.Success
+        integrator.force_stepfail = true
+        return
+    end
+    aₙ₊₁ = nlcache.u
+
+    u = ArrayPartition(
+        vₙ + dt * ((1 - γ) * aₙ + γ * aₙ₊₁),
+        uₙ + dt * vₙ + dt^2 / 2 * ((1 - 2β) * aₙ + 2β * aₙ₊₁)
```

Comment:
ad some `@..` to reduce allocations?

Reply:
This catches fire. `DimensionMismatch: array could not be broadcast to match destination`